test/e2e/aws_eip_for_nlb_test.go

```diff
@@ -0,0 +1,434 @@
+			}
+			if networkLoadBalancerParametersEIPAllocationStatusExist(ic) {
+				eipAllocationStatus = ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations
+			}
+		}
+
+		// Check if the eipAllocationSpec and eipAllocationStatus are equal.
+		if !reflect.DeepEqual(eipAllocationSpec, eipAllocationStatus) {
+			t.Logf("expected ingresscontroller eipAllocation status to be %q, got %q, retrying...", eipAllocationSpec, eipAllocationStatus)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("error verifying eipAllocation status for IngressController %q: %v", icName.Name, err)
+	}
+	return nil
+}
+
+// networkLoadBalancerParametersEIPAllocationSpecExists checks if NetworkLoadBalancerParameters exist in the IngressController spec.
+func networkLoadBalancerParametersEIPAllocationSpecExists(ic *operatorv1.IngressController) bool {
+	return ic.Spec.EndpointPublishingStrategy != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters != nil
+}
+
+// networkLoadBalancerParametersEIPAllocationStatusExist checks if NetworkLoadBalancerParameters exist in the IngressController
+// status.
+func networkLoadBalancerParametersEIPAllocationStatusExist(ic *operatorv1.IngressController) bool {
+	return ic.Status.EndpointPublishingStrategy != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters != nil &&
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.AWS.NetworkLoadBalancerParameters.EIPAllocations != nil
+}
+
+func createInvalidEIPAllocations(t *testing.T) []operatorv1.EIPAllocation {
+	publicSubnets, _, _ := processForFetchingSubnets(t)
+	var err error
+	invalidEIPAllocations := make([]operatorv1.EIPAllocation, len(publicSubnets))
+	var hexString string
+	for i := 0; i < len(publicSubnets); i++ {
+		if hexString, err = randomHex(9); err != nil {
+			return nil
+		}
+		invalidEIPAllocations[i] = operatorv1.EIPAllocation(fmt.Sprintf("%s-%s", "eipalloc", hexString[:17]))
+	}
+	return invalidEIPAllocations
+}
+
+func randomHex(n int) (string, error) {
+	bytes := make([]byte, n)
+	if _, err := random.Read(bytes); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes), nil
+}
+
+func processForFetchingSubnets(t *testing.T) (map[string]struct{}, *ec2.EC2, string) {
```

Comment:
This function is confusing to me. It's pretty ambiguous, returns seemingly unrelated objects, I'm not sure what `processForFetchingSubnets` really means.

Couple of things:
1. Why not create the EC2 service client in the top level `TestAWSEIPForNLB` and pass it around as a function parameter to `CreateAWSEIPs` and `createInvalidEIPAllocations`? You do this with `cleanupEIPAllocations`. Seems like building a new service client in each function is clunky.
2. If you do that above, you don't need this function. Just get the VPCID, and call `getPublicSubnets`.
3. If you want a helper for creating EC2 service client, you could call it `createEC2ServiceClient`.

Reply:
done